--- a/gymnasium_robotics/envs/multiagent_mujoco/mujoco_multi.py
+++ b/gymnasium_robotics/envs/multiagent_mujoco/mujoco_multi.py
@@ -209,7 +209,7 @@
             asset_path = os.path.join(
                 os.path.dirname(os.path.abspath(__file__)),
                 "assets",
-                f"many_segment_ant_{n_segs}_segments.auto.xml",
+                f"many_segment_ant_{n_segs}_segments.{os.getpid()}.auto.xml",
             )
             many_segment_ant.gen_asset(n_segs=n_segs, asset_path=asset_path)
             single_agent_env = gymnasium.make(
@@ -226,7 +226,7 @@
             asset_path = os.path.join(
                 os.path.dirname(os.path.abspath(__file__)),
                 "assets",
-                f"many_segment_swimmer_{n_segs}_segments.auto.xml",
+                f"many_segment_swimmer_{n_segs}_segments.{os.getpid()}.auto.xml",
             )
             many_segment_swimmer.gen_asset(n_segs=n_segs, asset_path=asset_path)
             single_agent_env = gymnasium.make(
